lib/hyacinth/adapters/preservation_adapter/fedora3/text_resource_properties.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module Adapters
+    module PreservationAdapter
+      class Fedora3::TextResourceProperties
+        include Fedora3::DatastreamMethods
+
+        def self.from(hyacinth_obj)
+          new(hyacinth_obj)
+        end
+
+        def initialize(hyacinth_obj)
+          @hyacinth_obj = hyacinth_obj
+        end
+
+        def to(fedora_obj)
+          return unless @hyacinth_obj.is_a? ::DigitalObject::Asset
+          ::DigitalObject::Asset::TEXT_RESOURCE_NAMES.each do |resource_name|
+            next unless @hyacinth_obj.send :"has_#{resource_name}_resource?"
+            resource = @hyacinth_obj.send :"#{resource_name}_resource"
+            next unless resource.preservable && resource.content_exists?
+            ensure_datastream(fedora_obj, resource_name, datastream_props_for(resource_name, resource))
+            datastream = fedora_obj.datastreams[resource_name]
+            content_digest = md5_digest(resource)
```

Comment:
@barmintor The resource object has a `#checksum` method on it that you can use instead.

Reply:
Marking resolved (now using the sha256)